core/jazz_acl/components/scm/bitbucket.js

```diff
@@ -0,0 +1,135 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const request = require("request");
+const logger = require("../logger.js");
+
+const addRepoPermissionInBitbucket = async (config, serviceInfo, policies) => {
+  let users_list = [];
+  await removeAllRepoUsers(config, serviceInfo);
+  for (const policy of policies.details) {
+    try {
+      const repoInfo = {
+        "repo_name": `${serviceInfo.domain}_${serviceInfo.service}`,
+        "repo_user": policy.userId,
+        "permission": `REPO_${policy.permission}`
+      };
+      if (config.PERMISSION_CATEGORIES.includes(policy.category)) {
+        await addPermsInBitbucketRepo(config, repoInfo);
+        users_list.push(policy.userId);
+      }
+    } catch (e) {
+      logger.error("Remove user failed for user : " + user.name);
+    }
+  }
+  logger.info("Updated users: " + users_list);
+  return users_list;
+};
+
+const addPermsInBitbucketRepo = async (config, repoInfo) => {
+  try {
+    let url = `${config.HOSTNAME}${config.REPO_BASE_API}${repoInfo.repo_name}${config.REPO_USER_PERM_API}${repoInfo.permission}&name=${repoInfo.repo_user}`;
```

Comment:
for variables that are not going to change try to use const

Reply:
done